rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,825 @@
+        """
+        Verifies two metadata dictionaries match in a set of required fields. If
+        a field is present in only one or neither of the two dictionaries, this
+        is considered a match.
+
+        This is primarily intended as a safety check, and does not conclusively
+        determine that two images are valid to append to together or are part of
+        the same series.
+
+        Args:
+            meta1: First metadata dictionary
+            meta2: Second metadata dictionary
+
+        Returns:
+            True if all keys that are present in both dictionaries have
+            equivalent values, False otherwise.
+
+        """
+        matchFields = ["Modality", "MagneticFieldStrength", "ImagingFrequency",
+                       "Manufacturer", "ManufacturersModelName",
+                       "InstitutionName", "InstitutionAddress",
+                       "DeviceSerialNumber", "StationName", "BodyPartExamined",
+                       "PatientPosition", "EchoTime",
+                       "ProcedureStepDescription", "SoftwareVersions",
+                       "MRAcquisitionType", "SeriesDescription", "ProtocolName",
+                       "ScanningSequence", "SequenceVariant", "ScanOptions",
+                       "SequenceName", "SpacingBetweenSlices", "SliceThickness",
+                       "ImageType", "RepetitionTime", "PhaseEncodingDirection",
+                       "FlipAngle", "InPlanePhaseEncodingDirectionDICOM",
+                       "ImageOrientationPatientDICOM", "PartialFourier"]
+
+        # If either field is None, skip and continue checking other fields
+        for field in matchFields:
+            value1 = meta1.get(field, None)
+            if value1 is None:
+                continue
+
+            value2 = meta2.get(field, None)
+            if value2 is None:
+                continue
+
+            if value1 != value2:
+                errorMsg = (f"Metadata doesn't match on field: {field} "
+                            f"(value 1: {value1}, value 2: {value2}")
+                return (False, errorMsg)
+
+        # These fields should not match between two images for a valid append
+        # (i.e., data should be different)
+        differentFields = [
+            "AcquisitionTime",  # should be collected at different times
+            "AcquisitionNumber",  # acquisition ID's should be different
+            ]
+
+        for field in differentFields:
+            value1 = meta1.get(field, None)
+            if value1 is None:
+                continue
+
+            value2 = meta2.get(field, None)
+            if value2 is None:
+                continue
+
+            if value1 == value2:
+                errorMsg = (f"Metadata matches (shouldn't) on field: {field} "
+                            f"(value 1: {value1}, value 2: {value2})")
+                return (False, errorMsg)
+
+        return (True, "")
+
+    def appendIncremental(self,
```

Comment:
Let's look at some ways to optimize performance for getIncremental and appendIncremental